### DISCUSSION TITLE
How to share with more people

### DISCUSSION THREAD

**Discussion by peter100u** [py5coding/py5generator#440] [Ideas]
As far as I know, it only runs on local machines. Good works deserve to be shared, so my question is, can it run on the browser so we can share it with more friends

**Comment by hx2A** [comment]
Hello, @peter100u !

I like your question, for a couple of reasons. I am happy to hear of your motivation to share your py5 work with others.

py5 is usually run on a local machine. It is possible to run py5 on a server with [JupyterHub](https://jupyter.org/hub), including free services like [binder](https://mybinder.org/). It can run in Google Colab but Colab has some weird issues that I can't deal with now. But anyway, when run on a server, the question becomes, how do you see it and how do you interact with it?

For a while py5 had a [ipython widget](https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html) called a SketchPortal that let you see the running Sketch and even responded to mouse and keyboard inputs. The frame rate wasn't perfect but it did in fact work. Unfortunately it broke a few releases ago because of changes in Jupyter. I have a idea for how to address this and will look into that later this year. If I got it to work again, it might be possible to do something interesting with [Mercury](https://runmercury.com/). That's on my list of things to play around with when I get a chance.

Finally, I'll point out you can always record a Sketch and make a video out of it. That probably isn't what you had in mind when you asked about sharing though. If sometime you do want to do that  you can save a sequence of frames using `py5_tools.save_frames()`.

**Reply by peter100u** [reply]
Thank you for your reply. I may prefer py5 to run in the browser, such as brython and skulpt, which can run Python code in the browser. However, I don't know their principles, which are simple and conducive to the dissemination of works and queues, similar to the principles of processing.js

**Reply by hx2A** [reply]
@peter100u , you should have a look at [Proceso](https://github.com/nickmcintyre/proceso/tree/main), a version of Processing that runs in a browser and uses Python. It was created by @nickmcintyre and internally uses the JavaScript library P5.

**Reply by villares** [reply]
Also, @peter100u, while I think Proceso has a great future ahead, if you want a quick hack with py5-like function names, you can try https://abav.lugaralgum.com/pyp5js/py5mode/ (based on Bernardo Fontes' pyp5js tool)
